server/src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -177,7 +178,7 @@ private static Timer maybeStartTimer(DfsProfiler profiler, DfsTimingType dtt) {
         return null;
     };
 
-    private static void executeKnnVectorQuery(SearchContext context) throws IOException {
+    static void executeKnnVectorQuery(SearchContext context) throws IOException {
```

Comment:
I think eagerly validating like this is OK. However, `KnnVectorQueryBuilder#doToQuery` should also validate as its possible to provide a knn query that isn't executed through DFS.

Reply:
done